apps/s_client.c

```diff
@@ -3553,4 +3556,72 @@ static char *base64encode (const void *buf, size_t len)
     return out;
 }
 
+/*
+ * Host dNS Name verifier: used for checking that the hostname is in dNS format 
+ * before setting it as SNI
+ */
+static int is_dNS_name(const char *host)
+{
+    const int MAX_LABEL_LENGTH = 63;
+
+    int i;
+    int isdnsname = 0;
+    int length = strlen(host);
+    int label_length = 0;
+    int all_numeric = 1;
+
+    /*
+     * XXX: Deviation from strict DNS name syntax, also check names with '_'
+     * Check DNS name syntax, any '-' or '.' must be internal,
+     * and on either side of each '.' we can't have a '-' or '.'.
+     *
+     * If the name has just one label, we don't consider it a DNS name.  This
+     * means that "CN=sometld" cannot be precluded by DNS name constraints, but
+     * that is not a problem.
```

Comment:
The second sentence of the second comment paragraph is about X.509 and does not apply here. Delete it.

Reply:
Removed